spot_rl_experiments/spot_rl/envs/nav_env.py

```diff
@@ -206,8 +228,60 @@ def get_success(self, observations):
             self.spot.set_base_velocity(0.0, 0.0, 0.0, 1 / self.ctrl_hz)
         return succ
 
+    def get_hand_xy_theta(self, use_boot_origin=False):
+        """
+        Much like spot.get_xy_yaw(), this function returns x,y,yaw of the hand camera instead of base such as in spot.get_xy_yaw()
+        Accepts the same parameter use_boot_origin of type bool like the function mentioned in above line, this determines whether the calculation is from the vision frame or robot'home
+        If true, then the location is calculated from the vision frame else from home/dock
+        Returns x,y,theta useful in head/hand based navigation used in Heurisitic Mobile Navigation
+        """
+        vision_T_hand = get_a_tform_b(
+            self.spot.robot_state_client.get_robot_state().kinematic_state.transforms_snapshot,
+            "vision",
+            "hand",
+        )
+        theta = quat_to_eulerZYX(vision_T_hand.rotation)[0]
+        point_in_global_2d = np.array([vision_T_hand.x, vision_T_hand.y])
+        return (
+            (point_in_global_2d[0], point_in_global_2d[1], theta)
+            if use_boot_origin
+            else self.spot.xy_yaw_global_to_home(
+                point_in_global_2d[0], point_in_global_2d[1], theta
+            )
+        )
+
+    def get_nav_observation_by_hand(self, goal_xy, goal_heading):
+        observations = {}
+
+        # Get visual observations
+        front_depth = self.msg_to_cv2(self.filtered_head_depth, "mono8")
+
+        front_depth = cv2.resize(
+            front_depth, (120 * 2, 212), interpolation=cv2.INTER_AREA
+        )
+        front_depth = np.float32(front_depth) / 255.0
+        # Add dimension for channel (unsqueeze)
+        front_depth = front_depth.reshape(*front_depth.shape[:2], 1)
+        observations["spot_right_depth"], observations["spot_left_depth"] = np.split(
+            front_depth, 2, 1
+        )
```

Comment:
Same variable is getting constantly updated. It might make sense to create a separate function `get_head_depth`

Reply:
made a new fn get_head_depth